pkg/tfshim/walk/walk_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package walk
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/schema"
+)
+
+var strSchema = (&schema.Schema{
+	Type:     shim.TypeString,
+	Optional: true,
+}).Shim()
+
+var testSchemaMap shim.SchemaMap = schema.SchemaMap{
+	"x": (&schema.Schema{
+		Type: shim.TypeMap,
+		Elem: (&schema.Resource{
+			Schema: schema.SchemaMap{
+				"y": strSchema,
+			},
+		}).Shim(),
+	}).Shim(),
+
+	"list": (&schema.Schema{
+		Type: shim.TypeList,
+		Elem: strSchema,
+	}).Shim(),
+
+	"batching": (&schema.Schema{
+		Type:     shim.TypeList,
+		MaxItems: 1,
+		Elem: (&schema.Resource{
+			Schema: schema.SchemaMap{
+				"send_after": strSchema,
+			},
+		}).Shim(),
+	}).Shim(),
+}
+
+func TestLookupSchemaPath(t *testing.T) {
+	s := testSchemaMap
+
+	type testCase struct {
+		name   string
+		path   SchemaPath
+		expect any
+	}
+
+	testCases := []testCase{
+		{
+			"single-nested block object",
+			NewSchemaPath().GetAttr("x"),
+			s.Get("x"),
+		},
+		{
+			"cannot do Element on an object",
+			NewSchemaPath().GetAttr("x").Element(),
```

Comment:
This fails; although NewSchemaPath().GetAttr("x") looks up a Schema with an Elem(), we can't drill down an Element(). The motivation for this is that values of this type would look like this:

```
v = {"x": {"y": "foo"}}
```

So drilling down to "foo" is done via `v["x"]["y"]` not `v["x"][index]["y"]`.